packages/react-select/src/Select.js

```diff
@@ -378,7 +379,8 @@ export default class Select extends Component<Props, State> {
 
         return isEqual(newSelectValue, lastSelectValue)
           && isEqual(newProps.inputValue, lastProps.inputValue)
-          && isEqual(newProps.options, lastProps.options);
+          && isEqual(newProps.options, lastProps.options)
+          && newProps.isOptionDisabled === lastProps.isOptionDisabled;
```

Comment:
If we're going to add functions to the memoization we should also include `isOptionSelected`, `getOptionLabel`, `getOptionValue`, `filterOption`, etc.

Reply:
I've these and some other comparisons which are needed for rebuilding menu options. Do you expect to have tests for each of these? I've added one for the original issue.